semanticscholar/ApiRequester.py

```diff
@@ -14,14 +14,16 @@
 
 class ApiRequester:
 
-    def __init__(self, timeout, debug) -> None:
+    def __init__(self, timeout, debug, with_retry: bool = True) -> None:
         '''
         :param float timeout: an exception is raised \
             if the server has not issued a response for timeout seconds.
         :param bool debug: enable debug mode.
+        :param bool with_retry: enable retry mode.
```

Comment:
maybe I don't fully understand the meaning of the max_results parameter. my purpose is to reduce the time it costs to call the API and fail fast so that users will not experience obvious delays. Thanks!